src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx

```diff
@@ -129,7 +127,50 @@ export const WorkspaceUpdater = () => {
       }
     }
     setDeleteWorkspaceModalVisible(false);
-    await application.navigateToApp('home');
+    if (http) {
+      await application.navigateToUrl(
+        http.basePath.remove(
+          application.getUrlForApp('home', {
+            path: '',
+            absolute: false,
+          })
+        )
+      );
+    }
+  };
+
+  const exitWorkspace = async () => {
+    let result;
+    try {
+      result = await workspaces?.client.exitWorkspace();
+    } catch (error) {
+      notifications?.toasts.addDanger({
+        title: i18n.translate('workspace.exit.failed', {
+          defaultMessage: 'Failed to exit workspace',
+        }),
+        text: error instanceof Error ? error.message : JSON.stringify(error),
+      });
+      return;
+    }
+    if (!result?.success) {
+      notifications?.toasts.addDanger({
+        title: i18n.translate('workspace.exit.failed', {
+          defaultMessage: 'Failed to exit workspace',
+        }),
+        text: result?.error,
+      });
+      return;
+    }
+    if (http) {
+      await application.navigateToUrl(
```

Comment:
Same here, `workspace.formatUrlWithWorkspaceId` can remove the workspace prefix (/w/${workspaceId}) if passed `workspaceId` is empty, basePath.remove will remove the serverBasePath as well, which will lead to bug.